src/Service/GroupsService.php

```diff
@@ -33,6 +37,59 @@ public function __construct(
         parent::__construct($repository, $mapper, $cache);
     }
 
+    /**
+     * @param CoreEntity $coreEntity
+     * @param string $groupType
+     * @param int|null $limit
+     * @param int $page
+     * @param string $ttl
+     * @param string $nullTtl
+     * @return Group[]
+     */
+    public function findByCoreEntityMembership(
+        CoreEntity $coreEntity,
+        string $groupType = 'Group',
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE,
+        $ttl = CacheInterface::NORMAL,
+        $nullTtl = CacheInterface::NORMAL
+    ): array {
+        if (!($coreEntity instanceof Programme || $coreEntity instanceof Collection || $coreEntity instanceof Gallery)) {
+            throw new InvalidArgumentException(sprintf(
+                'Called %s with an invalid type. Expected a Programme, Collection or Gallery but got "%s"',
+                debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2)[1]['function'],
+                \get_class($coreEntity)
+            ));
+        }
+
+        if (!\in_array($groupType, ['Group', 'Collection', 'Franchise', 'Gallery', 'Season'])) {
+            throw new InvalidArgumentException(sprintf(
+                'Called %s with an invalid type. Expected Group, Collection, Franchise, Gallery or Season but got "%s"',
+                debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2)[1]['function'],
```

Comment:
This debug_backtrace weirdness is because in other class this check is extracted out into a private method, but we want the exception message to say about "Called findByCoreEntityMembership with an invalid type".

If you're doing the throw the actual method instead of a helper method you can use `__FUNCTION__` to get the name of the current function instead of parsing the backtrace to find the calling function.